src/Tools/dotnet-monitor/ActivityExtensions.cs

```diff
@@ -7,7 +7,7 @@ namespace Microsoft.Diagnostics.Tools.Monitor
 {
     internal static class ActivityExtensions
     {
-        public static string GetSpanId(this Activity activity)
+        public static string? GetSpanId(this Activity activity)
         {
             switch (activity.IdFormat)
```

Comment:
Why is this nullable? It seems like every code path returns a non null string.

Reply:
`Activity.Id` can be null.

https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.activity.id?view=net-8.0

Reply:
I've updates https://github.com/dotnet/dotnet-monitor/issues/6929 to keep track of this.